core/src/layers/prometheus.rs

```diff
@@ -88,12 +88,21 @@ use crate::*;
 #[derive(Default, Debug, Clone)]
 pub struct PrometheusLayer {
     registry: Registry,
+    metric_bucket: PrometheusHistogramBucket,
 }
 
 impl PrometheusLayer {
     /// create PrometheusLayer by incoming registry.
     pub fn with_registry(registry: Registry) -> Self {
-        Self { registry }
+        Self {
+            registry,
+            metric_bucket: PrometheusHistogramBucket::default(),
+        }
+    }
+    /// set metric bucket.
+    pub fn metric_bucket(&mut self, metric_bucket: PrometheusHistogramBucket) -> &mut Self {
```

Comment:
It's better to not expose a new struct like `PrometheusHistogramBucket`. We can accept a `Vec<f64>` directly as input. For example:

```rust
pub fn request_duration_seconds_buckets(mut self, buckets: Vec<f64>) -> Self { .. }
```

Reply:
Is it better to return `Self` than `&mut Self` here?

Reply:
How about rename the Struct to PromethuesConfig? I prefer keep the method argument clean

Reply:
> How about rename the Struct to PromethuesConfig? I prefer keep the method argument clean

No, let's maintain a consistent configuration style throughout all the layers.

Reply:
LGTM